sdks/java/extensions/sql/zetasql/src/test/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSqlDialectSpecTest.java

```diff
@@ -3843,4 +3847,27 @@ public void testArrayAggZetasql() {
 
     pipeline.run().waitUntilFinish(Duration.standardMinutes(PIPELINE_EXECUTION_WAITTIME_MINUTES));
   }
+
+  @Test
+  public void testArrayConcatAggZetasql() {
+    String sql =
+        "WITH aggregate_example AS (SELECT [1,2] AS numbers  UNION ALL SELECT [3,4] AS numbers UNION ALL SELECT [5, 6] AS numbers) SELECT ARRAY_CONCAT_AGG(numbers) AS count_to_six_agg FROM aggregate_example";
+
+    ZetaSQLQueryPlanner zetaSQLQueryPlanner = new ZetaSQLQueryPlanner(config);
+    BeamRelNode beamRelNode = zetaSQLQueryPlanner.convertToBeamRel(sql);
+    PCollection<Row> stream = BeamSqlRelUtils.toPCollection(pipeline, beamRelNode);
+    PAssert.thatSingleton(stream)
+        .satisfies(
+            row -> {
+              Collection<Object> output = row.getArray("count_to_six_agg");
+              HashSet<Object> outputSet = new HashSet<Object>(output);
+
+              HashSet<Object> expectedOutputSet = Sets.newHashSet(1L, 2L, 3L, 4L, 5L, 6L);
+
+              assertThat("array_field", expectedOutputSet.equals(outputSet));
```

Comment:
Use `containsInAnyOrder` instead of building and comparing sets. It's simpler. http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#containsInAnyOrder(T...)